### PR TITLE
fix(text) escape value before rendering styles

### LINKED ISSUE #19: Escaped escape codes when overriding ValueStyle and value needs quoting
I was trying to override `log.ValueStyle` to add some colour, however due to it being quoted after the `ValueStyle.Render(val)` the added escape codes themselves end up escaped.

```go
package main

import (
    "os"

    "github.com/charmbracelet/lipgloss"
    "github.com/charmbracelet/log"
)

func main() {
    logger := log.New(
        log.WithOutput(os.Stderr),
         log.WithTimestamp()
    )
    log.ValueStyle = lipgloss.NewStyle().
        Foreground(lipgloss.AdaptiveColor{Light: "208", Dark: "192"})
    logger.Info("testing overriding global styles", "key", "val", "foo", true)
}
```

This results in the following output in the terminal

charmbracelet/log output

The expected output would be

expected charmbracelet/log output

I would guess that the fix would be to call `lipgloss.Render(val)` after the value has been escaped to prevent the lipgloss escape codes from being escaped as part of the value.

### PR DESCRIPTION
**Pull request by tombell**
Update to essentially render lipgloss styles for values after escaping and indenting.

- Add `l *looger` receiver to `writeEscapedForOutput` for access to `l.noStyles`
- Add `l *logger` receiver to `writeIdent` to be able to call the updated `writeEscapedForOutput`
- Update `writeEscapedForOutput` to call `ValueStyle.Render` after escaping the values, or if the values don't need need escaping

Fix #19

### CODE REVIEW

**Comment by aymanbagabas** [comment]
Thank you @tombell! Adding a test for this would be nice :)

**Comment by tombell** [comment]
@aymanbagabas just figuring out how the existing test broke first :)

**Comment by tombell** [comment]
@aymanbagabas trying to figure out the best approach to adding some test cases. Since [`noStyles`](https://github.com/charmbracelet/log/blob/main/log.go#L273) will always be `false` not sure how to get it to render with the escape codes to check the output. Do you have any suggestions?

**Comment by aymanbagabas** [comment]
@tombell you could modify the `*logger` struct `noStyles` value directly in the tests

```go
func TestStyles(t *testing.T) {
  var buf bytes.Buffer
  l := New(WithOutput(&buf)).(*logger)
  l.noStyles = false
  ...
}
```

**Comment by tombell** [comment]
@aymanbagabas believe I've got this sorted now, all pushed up.

File: text_test.go [hunk 1]
```diff
@@ -228,3 +229,78 @@ func TestTextFatal(t *testing.T) {
...
+	logger := New(WithOutput(&buf)).(*logger)
+	logger.noStyles = false
+	ValueStyle = lipgloss.NewStyle().Bold(true)
+	cases := []struct {
+		name     string
+		expected string
+		msg      string
+		kvs      []interface{}
+		f        func(msg interface{}, kvs ...interface{})
+	}{
```
Comment (aymanbagabas) [inline text_test.go:244]: Could you add a test case for a multiline value?

Comment (tombell) [reply text_test.go:244]: Just pushed up that, including having to update `writeIdent` to account for styling the values after escaping as well. I might look into extracting the post-styling after escaping into the escape function itself.

**Comment by tombell** [comment]
I believe I've refactored enough to simplify the solution.

**Pull request merged**